### ISSUE TITLE
Weakly typed helper issue related to output html rendering.

### ISSUE THREAD

**Issue by JaroslawWaliszko** [aspnet/Mvc#1979]
In ASP.NET MVC 5.2.2 version I've noticed minor mismatch in output html rendering between weakly and strongly typed html helpers, mainly:

* usage of `@Html.DropDownList("CarModelID", null, "-- select --", htmlAttributes...)` gives cropped output:

    ![image](https://cloud.githubusercontent.com/assets/1596462/6108150/11c2d940-b071-11e4-84cc-c257c8cd2dae.png)

* while `@Html.DropDownListFor(model => model.CarModelID, null, "-- select --", htmlAttributes...)` renders complete version (notice additional attributes):

    ![image](https://cloud.githubusercontent.com/assets/1596462/6107938/4dcd94a4-b06f-11e4-97ed-62bca50be723.png)

This situation reveals under certain circumstances, mainly:

* controller passes drop down items through `ViewBag`:

    ```
    public ActionResult Create()
    {
        ViewBag.CarModelID = new SelectList(db.CarModels, "CarModelID", "CarModelName");
    }
```
* model property is decorated by some validation attribute e.g.

    ```
    [Required]
    public int? CarModelID { get; set; }
```

More context, together with sample source code can be found [under this thread](https://github.com/JaroslawWaliszko/ExpressiveAnnotations/issues/46) (short outline [at the bottom](https://github.com/JaroslawWaliszko/ExpressiveAnnotations/issues/46#issuecomment-72981315) may also give some hints).

Regards

**Comment by yishaigalatzer** [comment]
@JaroslawWaliszko we are tracking MVC issue in http://aspnetwebstack.codeplex.com

**Comment by yishaigalatzer** [comment]
@dougbu can you please check what this does in this version of MVC

**Comment by dougbu** [comment]
@yishaigalatzer the behaviour with MVC 6 is the same.  this issue is another symptom of #600.

@JaroslawWaliszko MVC isn't finding a "weird object": it's finding the select list placed in `ViewBag`. but we cannot change how `ModelMetadata.FromStringExpression()` works, nor anything else about the priority of items in the `ViewBag` / `ViewData` dictionary compared to `ViewData.Model`.  the `ViewDataDictionary` infrastructure is too broadly used for such a change.  fortunately the problem here is specific to how `DropDownList()` and `ListBox()` operate when passed a `null` `selectList` value.

workarounds include:
- use `DropDownListFor()` instead; this option is mentioned in the [discussion](https://github.com/JaroslawWaliszko/ExpressiveAnnotations/issues/46) linked above
- use a different location in the `ViewBag` for the select list

e.g. in the controller,
``` c#
ViewBag.CarModelIDs = new SelectList(db.CarModels, "CarModelID", "CarModelName", car.CarModelID);
```
and in the views,
``` c#
@{ var selectList = (IEnumerable<SelectListItem>)(ViewBag.CarModelIDs);}
@Html.DropDownList("CarModelID", selectList, "-- select --", htmlAttributes: new { @class = "form-control" })
```

**Comment by JaroslawWaliszko** [comment]
@dougbu, @yishaigalatzer: I had tried to fix it but some of the unit tests were failing. I've read them, I know that the change would be problematic because of the broad usage of legacy implementation behaviors. 

What's more I don't even consider this issue as being problematic at all, since there are obviously approaches, which are clearly perceived as better and recommended over the usage reported in this thread. Nevertheless I've decided to post the issue here just to point it out and check what could be done in anything.

Regards
